wordpress-mobile/WordPress-Android
# [Jetpack Content Migration Flow] Add blank Welcome Screen (1)

Fixes Partially #17368
Depends on: #17386
`Ready for merge` after #17386

Adds a blank Welcome Screen after WordPress to Jetpack auto login, which is behind a temporary feature flag.

To test:

0. **Prerequisite:** Toggle the following feature flags `on` in `Wordpress/build.gradle` code:
   (it's not possible to do this from the app because we need to be logged in to switch feature flags):
   - Set `JETPACK_SHARED_LOGIN` to `true`
   - Set `JETPACK_PROVIDER_SYNC` to `true`
   - Set `JETPACK_MIGRATION_FLOW` to `true`
   - Press the `Sync Now` button that appears at the top right of the file editing view in Android Studio
1. Build & Install the WordPress app then Login
5. Build & Install the Jetpack app **using the same variant**
   (e.g. if WordPress was `wordpressWasabiDebug` , use `jetpackWasabiDebug`)
6. **Expect** to see the blank "Welcome to Jetpack!" screen
7. **Verify** that A new blank screen is displayed (will later contain the Welcome to Jetpack Screen)
   - A loading spinner shows initially
   - Then the WordPress to Jetpack logo is shown
9. **Expect** Closing and reopening Jetpack app to **not** show the blank "Welcome to Jetpack" screen
10. **Expect** Logging out and closing Jetpack app app to **not** show the blank  "Welcome to Jetpack" screen

### Screenshot
| <img width="314" src="https://user-images.githubusercontent.com/4588074/198836568-87df165f-eaf4-4265-866a-420c8993682c.png"> |
| - |

### Merge Strategy
1. Merge dependency PR first: #17386 into `trunk`
2. Remove `Not Ready for Merge` flag
3. Merge this PR

## Regression Notes
1. Potential unintended areas of impact
   Shared login flow.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
   Manual testing, updated tests in `SharedLoginResolverTest` for case with flag disabled.

3. What automated tests I added (or what prevented me from doing so)
   None for now - the logic is straightforward applied only when the feature flag is enabled in code.

PR submission checklist:
